### PR TITLE
SONARPY-642 Add standard library type definitions from Typeshed

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**

### CODE REVIEW

File: python-frontend/src/main/java/org/sonar/python/types/TypeShed.java [hunk 1]
```diff
@@ -117,6 +108,42 @@ public void visitFunctionDef(FunctionDef functionDef) {
...
+  }
+
+  public static Symbol standardLibrarySymbol(String stdLibModuleName, String fullyQualifiedName) {
+    Set<Symbol> librarySymbols = standardLibrarySymbols(stdLibModuleName);
+    return librarySymbols.stream().filter(s -> fullyQualifiedName.equals(s.fullyQualifiedName())).findFirst().orElse(null);
+  }
+
+  public static Map<String, Symbol> readTypeShedSymbols(String fileName, String moduleName) {
+    Map<String, Symbol> typeShedSymbols = new HashMap<>();
+    typeShedSymbols.put(NONE_TYPE, new ClassSymbolImpl(NONE_TYPE, NONE_TYPE));
```
[the remark this answers is not in the document]
Comment (guillaume-dequenne-sonarsource) [reply python-frontend/src/main/java/org/sonar/python/types/TypeShed.java]: Indeed, forgot to remove it! Thanks!

File: python-frontend/src/main/java/org/sonar/python/semantic/SymbolUtils.java [hunk 2]
```diff
@@ -89,9 +89,14 @@ public static String fullyQualifiedModuleName(String packageName, String fileNam
     Set<Symbol> globalSymbols = new HashSet<>();
     for (Symbol globalVariable : fileInput.globalVariables()) {
       if (globalVariable.kind() == Symbol.Kind.CLASS) {
-        globalSymbols.add(((ClassSymbolImpl) globalVariable).copyWithoutUsages());
+        String fullyQualifiedVariableName = globalVariable.fullyQualifiedName();
+        if (fullyQualifiedVariableName != null && fullyQualifiedVariableName.startsWith(fullyQualifiedModuleName)) {
+          globalSymbols.add(((ClassSymbolImpl) globalVariable).copyWithoutUsages());
+        } else {
+          globalSymbols.add(((ClassSymbolImpl) globalVariable).copyWithPrefix(fullyQualifiedModuleName));
```
[the remark this answers is not in the document]
Comment (guillaume-dequenne-sonarsource) [reply python-frontend/src/main/java/org/sonar/python/semantic/SymbolUtils.java]: Indeed it's not a correct approach, as the FQN stays the same. 

As discussed, I removed imported & builtin symbols from the project level symbol table for now. We should reintroduce them once we stop relying on FQN alone for single imported symbols (as we can have duplicate of those).

File: python-frontend/src/main/java/org/sonar/python/types/TypeShed.java [hunk 3]
```diff
@@ -117,6 +108,42 @@ public void visitFunctionDef(FunctionDef functionDef) {
     return fileInput.globalVariables().stream().filter(s -> s.fullyQualifiedName() != null).collect(Collectors.toMap(Symbol::fullyQualifiedName, Function.identity()));
   }
 
+  public static Set<Symbol> standardLibrarySymbols(String stdlibModuleName) {
+    if (!TypeShed.standardLibrarySymbols.containsKey(stdlibModuleName)) {
+      Map<String, Symbol> result = readTypeShedSymbols("stdlib/2and3/" + stdlibModuleName + ".pyi", stdlibModuleName);
+      if (result != null) {
+        standardLibrarySymbols.put(stdlibModuleName, result.values().stream().filter(s -> s.fullyQualifiedName() != null).collect(Collectors.toSet()));
```
[the remark this answers is not in the document]
Comment (guillaume-dequenne-sonarsource) [reply python-frontend/src/main/java/org/sonar/python/types/TypeShed.java:128]: As discussed, I kept it as it is for now.

File: sonar-python-plugin/pom.xml [hunk 4]
```diff
@@ -129,7 +129,7 @@
             <configuration>
               <rules>
                 <requireFilesSize>
-                  <maxsize>3300000</maxsize>
+                  <maxsize>4300000</maxsize>
                   <minsize>3000000</minsize>
```
Comment (andrea-guarino-sonarsource) [inline sonar-python-plugin/pom.xml:133]: Increase minimum size accordingly

**Pull request merged**